tmail-backend/mailbox/encrypted/cassandra/src/main/java/com/linagora/tmail/encrypted/cassandra/DeleteEncryptedProjectionHook.java

```diff
@@ -61,9 +62,9 @@ public Publisher<Void> notifyDelete(DeleteOperation deleteOperation) {
     }
 
     private Mono<Boolean> isMessageStillAccessible(DeletedMessageMailboxContext deleteContext) {
-        return Flux.from(messageIdManager.getMessagesReactive(ImmutableList.of(deleteContext.messageId()), FetchGroup.HEADERS, sessionProvider.createSystemSession(deleteContext.owner())))
+        return Flux.from(messageIdManager.getMessagesReactive(ImmutableList.of(deleteContext.messageId()), FetchGroup.MINIMAL, sessionProvider.createSystemSession(deleteContext.owner())))
             .map(MessageResult::getMailboxId)
-            .collectList()
+            .collect(Collectors.toSet())
             .map(listMailbox -> new HashSet<>(deleteContext.ownerMailboxes()).containsAll(listMailbox));
```

Comment:
Maybe `.filter` then `.hasElement` here is better than collecting 2 Sets and then iterating every item again?

Reply:
here is "containsAll" operation, 
I do not yet have an image of how the code looks like, can you give code examples?

Reply:
Here IMO `ownerMailboxes().containsAll(listMailbox)` = there is no element of `listMailbox` is not in `ownerMailboxes`.
Code example:
```
Flux.from(...)
       .filter(elementOfListMailbox -> !deleteContext.ownerMailboxes().contain(elementOfListMailbox))
       .hasElements()
       .map(bool -> !bool)
```

Reply:
Reversed logic here is a bit more hard to understand TBH, but I think this is good for performance.